elasticsearch/transport.py

```diff
@@ -335,6 +337,10 @@ def perform_request(self, method, url, headers=None, params=None, body=None):
             if isinstance(ignore, int):
                 ignore = (ignore,)
 
+        if headers is None:
+            headers = {}
+        headers["user-agent"] = "elasticsearch-py/%s (Python %s)" % (__versionstr__, python_version())
```

Comment:
this will always override the value in `headers` even if the user supplied their own, is that really what we want?

Reply:
That's a good question! I'll check in with other the other clients. What are your 2 cents on this? Did you came across a setup where `user-agent` was actively populated?

Reply:
Added PR #992 to address this.